backend/functions/src/index.ts

```diff
@@ -1,9 +1,73 @@
 import * as functions from 'firebase-functions';
+import {
+  decrementPostCount,
+  dislikePost,
+  getQuizWithSpecifiedRequirements,
+  getUserObj,
+  incrementPostCount,
+  likePost,
+  UserStatus,
+} from './tools';
 
-// // Start writing Firebase Functions
-// // https://firebase.google.com/docs/functions/typescript
-//
-// export const helloWorld = functions.https.onRequest((request, response) => {
-//   functions.logger.info("Hello logs!", {structuredData: true});
-//   response.send("Hello from Firebase!");
-// });
+export const getRandomQuiz = functions
+  .region('europe-west3')
+  .https.onRequest(async (req, res) => {
+    res.header('Access-Control-Allow-Origin', '*'); // update to match the domain you will make the request from
+    res.header(
+      'Access-Control-Allow-Headers',
+      'Origin, X-Requested-With, Content-Type, Accept'
+    );
+
+    try {
+      functions.logger.log(`rawBody = ${String(req.rawBody)}`);
+
+      const uid = req.body.data.uid as string;
```

Comment:
```suggestion
      if (!req.body || !req.body.data || req.body.data.uid !== 'string')
        handleError();;
      const { uid } = req.body.data;
```

Reply:
Thanks, I have added this refactoring with commits:
- [reduce amount of useless try/catch in backend](https://github.com/d9nchik/kpi_student/pull/6/commits/90cda29c69a430f5b4141dba6b67cd690a24dfd1)
- [remove redundant 'as' at backend/getRandomQuiz](https://github.com/d9nchik/kpi_student/pull/6/commits/2699399cba31d3da7c18b6cc53d77b1e0b4f8dd4)